psi4/driver/p4util/python_helpers.py

```diff
@@ -123,6 +123,120 @@ def pybuild_wavefunction(mol, basis=None):
 
 core.Wavefunction.build = pybuild_wavefunction
 
+
+@staticmethod
+def pyread_wavefunction(filename):
+    wfn_data = np.load(filename + '.npy').item()
+    
+    # variable type specific dictionaries to be passed into C++ constructor
+    wfn_matrix = wfn_data['matrix']
+    wfn_vector= wfn_data['vector']
+    wfn_dimension = wfn_data['dimension']
+    wfn_int = wfn_data['int']
+    wfn_string = wfn_data['string']
+    wfn_boolean = wfn_data['boolean']
+    wfn_float = wfn_data['float']
+
+    # reconstruct molecule from dictionary representation
+    wfn_molecule = wfn_data['molecule']
+    molecule = core.Molecule.from_dict(wfn_molecule)
+
+    # get basis set name and spherical harmonics boolean
+    basis_name = wfn_string['basisname']
+    if ".gbs" in basis_name:
+        basis_name = basis_name.split('/')[-1].replace('.gbs', '')
+
+    basis_puream = wfn_boolean['basispuream']
+    basisset = core.BasisSet.build(molecule, 'ORBITAL', basis_name, puream=basis_puream)
+    core.print_out('printing type of basis set\n')
+    core.print_out(str(type(basisset)))
+
+    # change some variables to psi4 specific data types (Matrix, Vector, Dimension)
+    # TODO: Make sure python's 'None' translates sensibly to C++
```

Comment:
I think None should translate to `nullptr` always here? Worth checking.

Reply:
Yeah, it does exactly that